packages/react-flight-dom-webpack/src/ReactFlightClientWebpackBundlerConfig.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+export type ModuleMetaData = {
+  id: string,
+  chunks: Array<Array<string>>,
+};
+
+export type Thenable = {
+  then(resolve: () => mixed, reject: (mixed) => mixed): mixed,
+  ...
+};
+
+export function requireModuleIfAvailable<T>(
+  moduleData: ModuleMetaData,
+): T | null {
+  let entry = require.cache[moduleData.id];
+  if (entry) {
+    return entry.exports.default;
+  }
+  // Ideally Webpack would let us inspect that all chunks have loaded and
```

Comment:
I chatted to @sokra about this and he says they can make it return the same Promise every time so we can put it in a WeakMap. Would this help us? My understanding is that this wouldn't be sufficient. If we just needed to track chunk completion status we could do it ourselves:

```js
__webpack_chunk_load__(chunk).then(() => { status[chunk] = true })
```

But that doesn't solve the problem of checking whether a particular has *already* loaded (either as the initial script on the page or as a part of some other request outside of Flight). I think that's what you want to check? In that case, I don't see a different way than exposing `installedChunks`. Maybe we can write a plugin that puts it on a global.

Reply:
Avoiding every possible async point is not a goal here. Just to minimize them. As discussed, I think the key here is to avoid the waterfall of requests that happens when each React component asks for them one by one. However, in this new approach we have all the info about all the chunks we're at high likelihood going to need soon without executing the React components. That's kind of the point because we should eagerly prefetch them regardless.

So I'm going to refactor this to eagerly load all the chunks and then keep track on our side whether it has already loaded. That way even if one needs to suspend, in that gap, we'll be able to collect a lot of newly resolved chunks at once and then continue.

Reply:
The key difference between this and the dynamic imports API is that loading the chunks doesn't immediately execute the module bodies. If we did, then executing these eagerly wouldn't be palatable.

So the deep Webpack integration is still useful.

Reply:
Note that chunk loading can be more than just loading javascript:
* adding stylesheets when using the mini-css-extract-webpack-plugin
* compiling wasm when using the sync wasm approach (deprecated in webpack 5)
* prefetching and preloading more chunks
* custom plugins would add additional logic here (e. g. I wrote a localization plugin which fetches localization data during chunk loading).

So direct `installedChunks` access would break when using more than JS.